src/main/java/com/sap/sgs/phosphor/fosstars/model/feature/oss/OssFeatures.java

```diff
@@ -216,4 +217,53 @@ private OssFeatures() {
    */
   public static final Feature<Boolean> SIGNS_ARTIFACTS
       = new BooleanFeature("If a project signs artifacts");
+
+  /**
+   * If OWASP Dependency plugin is used always to scan during the build.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_AS_MANDATORY_SCAN =
+      new BooleanFeature("If a project uses OWASP Dependency Check scan always");
+
+  /**
+   * If OWASP Dependency plugin is used always to report vulnerabilities during the build.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_AS_MANDATORY_REPORT =
+      new BooleanFeature("If a project uses OWASP Dependency Check to generate report always");
+
+  /**
+   * If OWASP Dependency plugin is used maybe during the build.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_AS_OPTIONAL_SCAN =
+      new BooleanFeature("If a project uses OWASP Dependency Check scan on specific occasions");
+
+  /**
+   * If OWASP Dependency plugin is used maybe to report vulnerabilities during the build.
+   */
+  public static final Feature<Boolean> USES_OWASP_DEPENDENCY_CHECK_AS_OPTIONAL_REPORT =
+      new BooleanFeature(
+          "If a project uses OWASP Dependency Check to generate report on specific occasions");
```

Comment:
Some combinations of those feature are invalid. For example, `USES_OWASP_DEPENDENCY_CHECK_AS_MANDATORY_SCAN == true and USES_OWASP_DEPENDENCY_CHECK_AS_OPTIONAL_SCAN == true` is invalid. The scores, which uses those features, must check for such invalid configurations. That would make them more complex. I would recommend to re-design the features to eliminate such invalid configurations. For example, you can add a feature that describes how the tool is used in the project. The feature can take three values: not used, mandatory or optional. If the tool runs, it always generates a report.

Reply:
Yes having 3 outcomes sounds better.
I will do the changes accordingly.